dev/EnvironmentManager/ChangeTracker/EnvironmentVariableChangeTrackerHelper.h

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+#pragma once
+
+#include <windows.h>
+#include <memory>
+#include <wil/result.h>
+#include <appmodel.h>
+#include <winrt/Windows.Management.Deployment.h>
+#include <Microsoft.Windows.System.EnvironmentManager.h>
+#include <AppModel.Identity.h>
+#include <Security.IntegrityLevel.h>
+
+
+using namespace winrt::Microsoft::Windows::System::implementation;
+
+static bool DoesChangeTrackingKeyExist()
+{
+    const auto subKey = std::filesystem::path{ LR"(Software\Microsoft\AppModel\RegistryWriteVirtualization\ExcludedKeys\HKEY_CURRENT_USER/Software/ChangeTracker)" };
+    wil::unique_hkey keyToChangeTracker{};
+    LSTATUS openResult{ RegOpenKeyEx(HKEY_LOCAL_MACHINE, subKey.c_str(), 0, KEY_READ | KEY_WOW64_64KEY, keyToChangeTracker.put()) };
+
+    LOG_IF_WIN32_ERROR_MSG(openResult, "Open key result to check if tracking exists: %i. Errors are treated as \"Do not track changes\"", openResult);
+
+    return openResult == ERROR_SUCCESS;
+}
+
+inline bool IsDllOnSystemAndRegistered()
+{
+
+    // Maybe a better way exists to switch between the test package family name and the production family name.
+    // This does not work because the test code and EM code are in two different processes.
+#if defined(__WINDOWSAPPRUNTIME_TEST_METADATA_H)
```

Comment:
>Maybe a better way exists to switch

It does ;-)

In your production code:

```
#include <dev\common\WindowsAppRuntime.VersionInfo.h>
...
const auto frameworkPackageFamily{ ::WindowsAppRuntime::VersionInfo::GetPackageFamilyName() };
```

and in your test call

```
#include <dev\common\WindowsAppRuntime.VersionInfo.h>
...
WindowsAppRuntime::VersionInfo::TestInitialize(::TP::WindowsAppRuntimeFramework::c_PackageFamilyName,
                                               ::TP::WindowsAppRuntimeMain::c_PackageFamilyName);

```

See `Unpackaged_SelfContained` in `test\common\Test_SelfContained.cpp` for an example.

WHY: Compile time hardwired for test vs production data is frowned upon given the difficulty using it appropriately. The better answer is to explicitly set a 'test override' when testing. This is how `MddBootstrapInitialize()` handled it and later more generalized to `WindowsAppRuntime.VersionInfo.h` because SelfContained and others needed the same sort of reindeer games.

Reply:
Awesome.  Thank you for that.  Code looks a lot less ugly. :)